From 4c20c27ee1b3d0c88ee5d6e41b88a5c94432dbcc Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@vmware.com>
Date: Thu, 20 Apr 2017 18:04:18 -0700
Subject: [PATCH] Include sysmacros.h directly as mandated by glibc-2.25.

glibc-2.25 generates a deprecation warning for using the 'major'
macro without including sysmacros.h.
https://sourceware.org/glibc/wiki/Release/2.25

This breaks Fedora builds of open-vm-tools because Fedora 26 is
switching to glibc-2.25.
---
 lib/wiper/wiperPosix.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/wiper/wiperPosix.c b/lib/wiper/wiperPosix.c
index d389eee..b41bd72 100644
--- a/lib/wiper/wiperPosix.c
+++ b/lib/wiper/wiperPosix.c
@@ -1,5 +1,5 @@
 /*********************************************************
- * Copyright (C) 2004-2016 VMware, Inc. All rights reserved.
+ * Copyright (C) 2004-2017 VMware, Inc. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU Lesser General Public License as published
@@ -30,6 +30,9 @@
 #include <stdio.h>
 #include <sys/stat.h>
 #if defined(__linux__) || defined(sun)
+# if defined(__linux__)
+#  include <sys/sysmacros.h>
+# endif
 # include <sys/vfs.h>
 #elif defined(__FreeBSD__) || defined(__APPLE__)
 # include <sys/param.h>
@@ -44,7 +47,6 @@
 
 #include "vmware.h"
 #include "wiper.h"
-#include "util.h"
 #include "str.h"
 #include "strutil.h"
 #include "fileIO.h"
